toddredmond/homework.md

```diff
@@ -0,0 +1,161 @@
+
+### Homework #1
+(Due Monday 1/12)
+
+You need to turn in this homework by cloning this repo, making a new branch with your solutions, pushing to github, and issuing a pull request.
+For details, follow the instructions [here](http://portlandcodeschool.github.io/jse/2015/01/07/command-line-and-git-slides/#/14).
+
+Please also read the [learning objectives](objectives.md) for this week.
+
+---
+
+**1)** (_Difficulty: mixed_)
+Predict the output of each of the following expressions, and explain the rules which determine it.
+If the output depends on the value of x, identify the conditions when the expression will output true.  Assume the cases are independent, and x is reset to an unknown value before each.
+
+Some of these are tricky!  Don't trust your first instinct.  
+
```

Comment:
Question #1 in general: Nice work! You managed to suss most of these out, and the ones that weren't quite there were mostly from edge cases. In particular, remember that NaN is not equal to anything, including itself, so in situations like 1q if x has the value of NaN you can see some pretty unexpected results. There are a couple of other edge cases that I'd be happy to chat about if you like, in particular in c, d, m, and n.